Build waku error

See https://github.com/himself65/waku/tree/experimental/bunchee

```json
{
  "name": "waku",
  "description": "⛩️ The minimal React framework",
  "version": "0.19.0-beta.2",
  "publishConfig": {
    "tag": "next"
  },
  "type": "module",
  "author": "Daishi Kato",
  "repository": {
    "type": "git",
    "url": "https://github.com/dai-shi/waku.git",
    "directory": "packages/waku"
  },
  "exports": {
    "./package.json": "./package.json",
    ".": {
      "types": "./dist/main.d.ts",
      "default": "./dist/main.js"
    },
    "./config": {
      "types": "./dist/config.d.ts",
      "default": "./dist/config.js"
    },
    "./prd": {
      "types": "./dist/prd.d.ts",
      "default": "./dist/prd.js"
    },
    "./dev": {
      "types": "./dist/dev.d.ts",
      "default": "./dist/dev.js"
    },
    "./node-loader": {
      "types": "./dist/node-loader.d.ts",
      "default": "./dist/node-loader.js"
    },
    "./client": {
      "types": "./dist/client.d.ts",
      "default": "./dist/client.js"
    },
    "./server": {
      "types": "./dist/server.d.ts",
      "default": "./dist/server.js"
    },
    "./router/client": {
      "types": "./dist/router/client.d.ts",
      "default": "./dist/router/client.js"
    },
    "./router/server": {
      "types": "./dist/router/server.d.ts",
      "default": "./dist/router/server.js"
    }
  },
  "bin": {
    "waku": "./dist/bin/cli.js"
  },
  "files": [
    "src",
    "dist"
  ],
  "scripts": {
    "dev": "bunchee -w",
    "compile": "rm -rf dist *.tsbuildinfo && bunchee && cp ../../README.md ."
  },
  "license": "MIT",
  "engines": {
    "node": ">=18.16.0"
  },
  "dependencies": {
    "@hono/node-server": "1.4.0",
    "@swc/core": "1.3.102",
    "@vitejs/plugin-react": "4.2.1",
    "hono": "3.11.12",
    "vite": "5.0.10"
  },
  "devDependencies": {
    "bunchee": "^4.4.0",
    "rollup": "^4.9.2"
  },
  "peerDependencies": {
    "react": "18.3.0-canary-c5b937576-20231219",
    "react-dom": "18.3.0-canary-c5b937576-20231219",
    "react-server-dom-webpack": "18.3.0-canary-c5b937576-20231219"
  }
}
```

```shell
/Users/himself65/Library/Caches/fnm_multishells/39569_1705274616742/bin/pnpm run dev

> waku@0.19.0-beta.2 dev /Users/himself65/Code/waku/packages/waku
> bunchee -w

⚠️ Cannot find source file for bin/waku
Watching assets in /Users/himself65/Code/waku/packages/waku...
error TS6379: Composite projects may not disable incremental compilation.
src/client.ts(17,29): error TS6307: File '/Users/himself65/Code/waku/packages/waku/src/lib/renderers/utils.ts' is not listed within the file list of project ''. Projects must list all files or use an 'include' pattern.
  File is ECMAScript module because '/Users/himself65/Code/waku/packages/waku/package.json' has field "type" with value "module"

error TS6379: Composite projects may not disable incremental compilation.

error TS6379: Composite projects may not disable incremental compilation.
src/router/server.ts(4,31): error TS6307: File '/Users/himself65/Code/waku/packages/waku/src/server.ts' is not listed within the file list of project ''. Projects must list all files or use an 'include' pattern.
  The file is in the program because:
    Imported via '../server.js' from file '/Users/himself65/Code/waku/packages/waku/src/router/server.ts'
    Imported via '../server.js' from file '/Users/himself65/Code/waku/packages/waku/src/router/server.ts'
  File is ECMAScript module because '/Users/himself65/Code/waku/packages/waku/package.json' has field "type" with value "module"
src/router/server.ts(6,32): error TS6307: File '/Users/himself65/Code/waku/packages/waku/src/client.ts' is not listed within the file list of project ''. Projects must list all files or use an 'include' pattern.
  File is ECMAScript module because '/Users/himself65/Code/waku/packages/waku/package.json' has field "type" with value "module"
src/router/server.ts(13,8): error TS6307: File '/Users/himself65/Code/waku/packages/waku/src/router/common.ts' is not listed within the file list of project ''. Projects must list all files or use an 'include' pattern.
  The file is in the program because:
    Imported via './common.js' from file '/Users/himself65/Code/waku/packages/waku/src/router/server.ts'
    Imported via './common.js' from file '/Users/himself65/Code/waku/packages/waku/src/router/server.ts'
  File is ECMAScript module because '/Users/himself65/Code/waku/packages/waku/package.json' has field "type" with value "module"
src/router/server.ts(15,26): error TS6307: File '/Users/himself65/Code/waku/packages/waku/src/lib/utils/path.ts' is not listed within the file list of project ''. Projects must list all files or use an 'include' pattern.
  File is ECMAScript module because '/Users/himself65/Code/waku/packages/waku/package.json' has field "type" with value "module"

error TS6379: Composite projects may not disable incremental compilation.

error TS6379: Composite projects may not disable incremental compilation.
src/router/client.ts(23,53): error TS6307: File '/Users/himself65/Code/waku/packages/waku/src/client.ts' is not listed within the file list of project ''. Projects must list all files or use an 'include' pattern.
  File is ECMAScript module because '/Users/himself65/Code/waku/packages/waku/package.json' has field "type" with value "module"
src/router/client.ts(29,8): error TS6307: File '/Users/himself65/Code/waku/packages/waku/src/router/common.ts' is not listed within the file list of project ''. Projects must list all files or use an 'include' pattern.
  The file is in the program because:
    Imported via './common.js' from file '/Users/himself65/Code/waku/packages/waku/src/router/client.ts'
    Imported via './common.js' from file '/Users/himself65/Code/waku/packages/waku/src/router/client.ts'
  File is ECMAScript module because '/Users/himself65/Code/waku/packages/waku/package.json' has field "type" with value "module"

error TS6379: Composite projects may not disable incremental compilation.
src/dev.ts(1,32): error TS6307: File '/Users/himself65/Code/waku/packages/waku/src/lib/middleware/hono-dev.ts' is not listed within the file list of project ''. Projects must list all files or use an 'include' pattern.
  File is ECMAScript module because '/Users/himself65/Code/waku/packages/waku/package.json' has field "type" with value "module"
src/dev.ts(2,35): error TS6307: File '/Users/himself65/Code/waku/packages/waku/src/lib/middleware/connect-dev.ts' is not listed within the file list of project ''. Projects must list all files or use an 'include' pattern.
  File is ECMAScript module because '/Users/himself65/Code/waku/packages/waku/package.json' has field "type" with value "module"
src/dev.ts(5,23): error TS6307: File '/Users/himself65/Code/waku/packages/waku/src/lib/builder/build.ts' is not listed within the file list of project ''. Projects must list all files or use an 'include' pattern.
  File is ECMAScript module because '/Users/himself65/Code/waku/packages/waku/package.json' has field "type" with value "module"

error TS6379: Composite projects may not disable incremental compilation.

error TS6379: Composite projects may not disable incremental compilation.
src/prd.ts(1,32): error TS6307: File '/Users/himself65/Code/waku/packages/waku/src/lib/middleware/hono-prd.ts' is not listed within the file list of project ''. Projects must list all files or use an 'include' pattern.
  File is ECMAScript module because '/Users/himself65/Code/waku/packages/waku/package.json' has field "type" with value "module"
src/prd.ts(2,35): error TS6307: File '/Users/himself65/Code/waku/packages/waku/src/lib/middleware/connect-prd.ts' is not listed within the file list of project ''. Projects must list all files or use an 'include' pattern.
  File is ECMAScript module because '/Users/himself65/Code/waku/packages/waku/package.json' has field "type" with value "module"
```

--- a/src/build-config.ts
+++ b/src/build-config.ts
@@ -186,6 +186,20 @@ async function buildInputConfig(
   const typesPlugins = [...commonPlugins, inlineCss({ skip: true })]
 
   if (useTypeScript) {
+    const enableIncrementalWithoutBuildInfo =
+      tsCompilerOptions.incremental && !tsCompilerOptions.tsBuildInfoFile
+    const incrementalOptions = enableIncrementalWithoutBuildInfo
+      ? {
+          incremental: false,
+        }
+      : undefined
+    const compositeOptions =
+      tsCompilerOptions.composite && enableIncrementalWithoutBuildInfo
+        ? {
+            composite: false,
+          }
+        : undefined
+
     const { options: overrideResolvedTsOptions }: any =
       await convertCompilerOptions(cwd, {
         declaration: true,
@@ -203,11 +217,9 @@ async function buildInputConfig(
           : undefined),
         // error TS5074: Option '--incremental' can only be specified using tsconfig, emitting to single
         // file or when option '--tsBuildInfoFile' is specified.
-        ...(tsCompilerOptions.incremental && !tsCompilerOptions.tsBuildInfoFile
-          ? {
-              incremental: false,
-            }
-          : undefined),
+        ...incrementalOptions,
+        // error TS6379: Composite projects may not disable incremental compilation.
+        ...compositeOptions,
       })
 
     const dtsPlugin = (

--- a/test/integration.test.ts
+++ b/test/integration.test.ts
@@ -846,8 +846,6 @@ const testCases: {
   {
     name: 'ts-composite',
     dir: 'monorepo/packages/package',
-    skip: true,
-    args: [],
     async expected(dir) {
       expect(await existsFile(join(dir, './dist/index.js'))).toBe(true)
       expect(await existsFile(join(dir, './dist/index.d.ts'))).toBe(true)

